### PR TITLE
Define CKD_SHA256_KDF transformation

### PR DESCRIPTION
**Pull request by freedge**
Define CKD_SHA256_KDF transformation to be used with CKM_ECDH1_DERIVE.

Some HSM with FIPS restriction will refuse to derive keys with CKD_NULL. CKD_SHA256_KDF will do fine though.

Unfortunately this is not implemented on softHSM
(https://github.com/opendnssec/SoftHSMv2/pull/599) so I provide no test. This was tested fine against Thales DPOD.

### CODE REVIEW

File: cryptoki/src/mechanism/elliptic_curve.rs
Comment (wiktor-k) [inline cryptoki/src/mechanism/elliptic_curve.rs:99]:
```suggestion
    /// enough bytes to fill the `CKA_VALUE_LEN` of the derived key.
```

**Comment by wiktor-k** [comment]
I think it looks good. 👍 

Unfortunately due to a new Rust version the lints started to pop up.

We could fix them in a similar way as in the tpm repo. What do you think @ionut-arm ?

**Review by wiktor-k — DISMISSED** [review]

**Comment by keldonin** [comment]
@freedge if that helps, the CI bot seems to work now, after the merge of PR #218. You might want to sync up your branch on your repo with the upstream one, to incorporate the latest fixes and get through these issues. I experienced the same and that solved the Execute CI script failures.

File: cryptoki/src/mechanism/elliptic_curve.rs
Comment (ionut-arm) [inline cryptoki/src/mechanism/elliptic_curve.rs:96]:
```suggestion
    /// The key derivation function based on sha256 as defined in the ANSI X9.63 standard. The
```
I'd recommend this change to make the reference clearer. If you'd like to put in a link to the standard that'd be even nicer 😉

File: cryptoki/src/mechanism/elliptic_curve.rs
Comment (ionut-arm) [inline cryptoki/src/mechanism/elliptic_curve.rs:100]:
```suggestion
    pub fn sha256() -> Self {
```
I think SHA256 is actually specific enough here, since the type makes it clear it's a KDF.

**Review by ionut-arm — COMMENTED** [review]
Thanks for the patch!

**Comment by hug-dev** [comment]
If you rebase and address @ionut-arm 's comments here we can easily get it merged :) !

**Comment by freedge** [comment]
let's close that for the moment as I don't have time or interest to work on it.
thanks for your project though!

**Comment by hug-dev** [comment]
Understood! Was a shame to not merge it after all your effort so pushed the same + the fix [here](https://github.com/parallaxsecond/rust-cryptoki/pull/239), hopefully we can get it merged soon and close the corresponding issue :)

**Pull request closed without merging**
